tree/20_cfe_basics/common.cf

```diff
@@ -328,3 +328,64 @@ bundle agent ncf_classes_copy(source_prefix, destination_prefix)
       "destination_not_empty"      not => strcmp("${destination_prefix}", "");
       "destination_exists"         and => { "destination_defined", "destination_not_empty" };
 }
+
+## Utility bundles to call sub methods
+# take care of dry-run and method_id management, plus allows copying the outcome
+
+bundle agent call_method(method) {
+  vars:
+    !pass1::
+      # store initial value
+      "should_report" string => "${report_data.should_report}";
+    pass3::
+      "method_id"  string => "${report_data.method_id}";
+      # store value with new method_id
+      "call_method_data.${method_id}_${method}" string => "${should_report}";
```

Comment:
It should be clearly stated that method and method_id can only use characters valids in classes name